channel.go

```diff
@@ -113,6 +113,10 @@ type ChannelOptions struct {
 	// Handler is an alternate handler for all inbound requests, overriding the
 	// default handler that delegates to a subchannel.
 	Handler Handler
+
+	// Dialer is optional factory method which can be used for overriding
+	// outbound connections for things like TLS handshake
+	Dialer func(ctx context.Context, hostPort string) (net.Conn, error)
```

Comment:
Should this function match the signature of `net.DialContext`?

Reply:
Didn't see much value in it originally as network is hardcoded to "tcp" all over the place.  Here is how it looks like if I make signature look the same.  Let me know what you think.